<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="viewport" content="width=device-width, initial-scale=1">



<title>Dirichlet adaptive shrinkage</title>

<link href="site_libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>




<link rel="stylesheet" href="cosmo.css" type="text/css" />

</head>

<body>


<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <link href="https://fonts.googleapis.com/css?family=Open+Sans:100,300,400,600" rel="stylesheet" type="text/css">
    <link href="../docs/cosmo.css" rel="stylesheet">
    <link href="../docs/cosmo.min.css" rel="stylesheet">
</head>
<script src="https://ajax.googleapis.com/ajax/libs/jquery/1.12.4/jquery.min.js"></script>
<script src="../docs/bootstrap/js/bootstrap.min.js"></script>

<body>

<div class = "nav" align = "middle">
<div class= "p1"></div>
<p>
  <a class="btn btn-large btn btn-primary" type="button" href = "Logolas_home.html">HOME</a>
  &nbsp &nbsp
  <a class="btn btn-large btn btn-primary" type="button" href = "Gallery.html">GALLERY</a>
  &nbsp &nbsp
  <a class="btn btn-large btn btn-primary" type="button" href = "https://github.com/kkdey/Logolas">SOFTWARE</a>
  &nbsp &nbsp
  <a class="btn btn-large btn btn-primary" type="button" href = "workflow.html">VIGNETTES</a>
  &nbsp &nbsp
  <a class="btn btn-large btn btn-primary" type="button" href = "license.html">LICENSE</a>
   &nbsp &nbsp
  <a class="btn btn-large btn btn-primary" type="button" href = "https://github.com/kkdey/Logolas-pages">GITHUB</a>
 </p>
</div>
<!-- <div class="nav">
  <a href="index.html">Home page</a> &nbsp &nbsp
  <a href="musings.html">Musings</a> &nbsp &nbsp
  <a href="town.html">My town</a> &nbsp &nbsp
  <a href="links.html">Links</a>
</div> -->

<!-- Main content -->
</body>
</html>


<h1 class="title toc-ignore">Dirichlet adaptive shrinkage</h1>


<div id="TOC">
<ul>
<li><a href="#idea-of-dirichlet-adaptive-shrinkage-dash">Idea of Dirichlet Adaptive Shrinkage (dash)</a></li>
<li><a href="#model">Model</a><ul>
<li><a href="#default-parameters">Default parameters</a></li>
</ul></li>
<li><a href="#example-illustration-of-dash">Example Illustration of dash</a></li>
<li><a href="#session-information">Session information</a></li>
</ul>
</div>

<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
<!-- Update knitr chunk options -->
<!-- Insert the date the file was last updated -->
<p><strong>Last updated:</strong> 2017-11-14</p>
<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
<p><strong>Code version:</strong> 35656d9</p>
<!-- Add your analysis here -->
<div id="idea-of-dirichlet-adaptive-shrinkage-dash" class="section level2">
<h2>Idea of Dirichlet Adaptive Shrinkage (dash)</h2>
<p>Typically logo plots are used to represent the position weight matrix (PWM). However, the PWM is often generated from underlying position frequency matrix (PFM). PFM stores the compositional data - the frequencies of the symbols at each position. Given a PFM matrix however, we also have knowledge of the scale of the frequencies for each position. This additional information can be used to get a better logo plot representation compared to the usual PWM-based logo plots.</p>
<p>Suppose for a specific position in a sequence, the compositional data is <span class="math display">\[ (A, C, G, T) : = (6, 1, 2, 1) \]</span> and in another case, it is <span class="math display">\[ (A, C, G, T) : = (600, 100, 200, 100). \]</span>. We assume that the background probability is equal for the four bases. If we transform the positional frequencies into PWM by taking the sample proportion, then the estimated PWM would be same for the two cases and would correspond to the same logo plot representation. However, in the second case, we have a total frequency of 1000 for all symbols while for the first case, it is only 10. Hence, it makes sense to shrink the PWM estimate to the background probability more strongly in the first case than that in the second case.</p>
<p>This is what we accomplish in an adaptive fashion using <code>dash</code> (Dirichlet Adaptive Shrinkage). We discuss the model formulation next.</p>
</div>
<div id="model" class="section level2">
<h2>Model</h2>
<p>Assume that there are <span class="math inline">\(L\)</span> constituents in the compositional mix. <span class="math inline">\(L\)</span> equals <span class="math inline">\(4\)</span> (corresponding to A,C, G and T bases) for the DNa sequence motif data and <span class="math inline">\(20\)</span> corresponding to the amino acids for the protein sequence data.</p>
<p>Suppose there are <span class="math inline">\(L\)</span> categories and <span class="math inline">\(n\)</span> positions. We model these compositional counts vectors as follows</p>
<p><span class="math display">\[ (c_{n1}, c_{n2}, \cdots, c_{nL}) \sim Mult \left ( c_{n+} : p_{n1}, p_{n2}, \cdots, p_{nL} \right ) \]</span></p>
<p>where <span class="math inline">\(c_{n+}\)</span> is the total frequency of the different constituents of the compositional data observed for the <span class="math inline">\(n\)</span> th base. <span class="math inline">\(p_{nl}\)</span> here represents the compositional probability for the <span class="math inline">\(l\)</span> th base in the position <span class="math inline">\(n\)</span>. We have</p>
<p><span class="math display">\[ p_{nl} &gt;= 0 \hspace {1 cm} \sum_{l=1}^{L} p_{nl} = 1 \]</span></p>
<p>We choose the Dirichlet prior distribution on the compositional probability vector <span class="math inline">\((p_{n1}, p_{n2}, \cdots, p_{nL})\)</span>. In order to perform adaptive shrinkage, we assume a mixture of known Dirichlet priors, each having mean to be the background mean probability vector ( <em>{1}, </em>{2}, , _{L} ), but with varying amounts of concentration, which need to be estimated along with the unknown mixture proportions from the data. <span class="math display">\[ \left ( p_{n1}, p_{n2}, \cdots, p_{nL} \right ) : = \sum_{k=1}^{K} \pi_{k} Dir \left (\alpha_{k} \mu_{1}, \alpha_{k} \mu_{2}, \cdots, \alpha_{k} \mu_{L} \right ) \hspace {1 cm} \alpha_{k} &gt; 0 \hspace{1 cm} \sum_{l=1}^{L} \mu_{l} = 1 \]</span> We assume a prior of <span class="math inline">\(\pi_{k}\)</span> to be Dirichlet</p>
<p><span class="math display">\[ f(\pi) : = \prod_{k=1}^{K} {\pi_{k}}^{\lambda_{k}-1} \]</span></p>
<div id="default-parameters" class="section level3">
<h3>Default parameters</h3>
<p>We choose a default set of <span class="math inline">\(\alpha_{k}\)</span> to be <span class="math inline">\((Inf, 100, 50, 20, 10, 2, 1, 0.1, 0.01)\)</span>. In this case <span class="math inline">\(\alpha_{k}=Inf\)</span> corresponds essentially to point mass at the prior mean or background mean ( <em>{1}, </em>{2}, , _{L} ), and then the subsequent choices of <span class="math inline">\(\alpha_{k}\)</span> have lower degree of concentration. <span class="math inline">\(\alpha_{k} = 1\)</span> corresponds to the most uniform scenario, whereas <span class="math inline">\(\alpha_{k} &lt; 1\)</span> correspond to cases with probability masses at the edges of the simplex but with the mean at the prior mean. The latter components would direct the points close to the corners towards the corners and away from the center, resulting in clearer separation of the points closer to the mean with the ones away from it.</p>
<p>We choose the default prior amount of shrinkage of <span class="math inline">\(\pi_{k}\)</span>, namely <span class="math inline">\(\lambda_{k}\)</span> to be <span class="math inline">\(\left( 1, 1, 1, 1, \cdots, 1 \right )\)</span>. The user may want to increase the weight on the first term (corresponding to ( _{k} = Inf ) ) to enforce stronger shrinkage.</p>
</div>
</div>
<div id="example-illustration-of-dash" class="section level2">
<h2>Example Illustration of dash</h2>
<p>We use the example of transcription factor, Aryl hydrocarbon receptor, to show the application of dash. Its PFM has 9 positions and each position has 154 counts, which is considerably large and hence the PWM computed for this data is reliable.</p>
<pre class="r"><code>library(Logolas)
library(grid)
color_profile = list(&quot;type&quot; = &quot;per_row&quot;, 
                     &quot;col&quot; = RColorBrewer::brewer.pal(4,name =&quot;Spectral&quot;))
scale1=function(x){return(x/sum(x))}
getpwm=function(pfm,bg=NULL,type=&#39;dash&#39;){
  if(is.null(bg)){bg=rep(1/nrow(pfm),nrow(pfm))}
  pfm=as.matrix(pfm)
  if(type==&#39;pse&#39;){
    pseudo=sqrt(dim(pfm)[2])*bg
    pwm=apply((pfm+pseudo),2,scale1)
  }else if(type==&#39;prop&#39;){
    pwm=apply(pfm,2,scale1)
  }else if(type==&#39;dash&#39;){
    pwm=dash(pfm,optmethod = &#39;mixEM&#39;,mode = bg)$posmean
  }else{pwm=NULL}
  if(nrow(pwm)==4){rownames(pwm)=c(&#39;A&#39;,&#39;C&#39;,&#39;G&#39;,&#39;T&#39;)}
  if(nrow(pwm)==20){rownames(pwm)=c(&#39;A&#39; ,&#39;R&#39;,&#39;N&#39;,&#39;D&#39;,   &#39;C&#39; ,  &#39;Q&#39;,   &#39;E&#39; ,  &#39;G&#39;,   &#39;H&#39; ,  &#39;I&#39;,   &#39;L&#39; , &#39;K&#39;  , &#39;M&#39; ,  &#39;F&#39;,   &#39;P&#39; ,  &#39;S&#39;  , &#39;T&#39; ,  &#39;W&#39;,   &#39;Y&#39;,   &#39;V&#39;)}
  colnames(pwm)=1:dim(pfm)[2]
  return(pwm)
}
#sampling function
dashsample=function(pfm,n){
  p=apply(pfm,2,function(x){
    sam=sample(rep(c(&#39;A&#39;,&#39;C&#39;,&#39;G&#39;,&#39;T&#39;),x),n,replace = F)
    n_a=sum(sam==&#39;A&#39;)
    n_g=sum(sam==&#39;G&#39;)
    n_c=sum(sam==&#39;C&#39;)
    n_t=sum(sam==&#39;T&#39;)
    return(c(n_a,n_c,n_g,n_t))
  })
  rownames(p)=c(&#39;A&#39;,&#39;C&#39;,&#39;G&#39;,&#39;T&#39;)
  return(p)
}
#pfm matrix
pfm=cbind(c(41,18,56,39),c(11,12,35,96),c(22,44,21,67),c(3,1,146,4),c(1,150,1,2),c(3,1,149,1),c(0,3,1,150),c(0,0,154,0),c(43,67,16,28))
rownames(pfm)=c(&#39;A&#39;,&#39;C&#39;,&#39;G&#39;,&#39;T&#39;)
colnames(pfm)=1:ncol(pfm)</code></pre>
<p>We use the PWM thus computed as true position weights and present a sequence logo for this matrix.</p>
<pre class="r"><code>logomaker(pfm,color_profile = color_profile,pop_name = &#39;True Sequence logo&#39;)</code></pre>
<p><img src="figure/dashtutor2.Rmd/unnamed-chunk-2-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>We do a sampling without replacement to simulate the cases when the number of counts is small. For each position, we sample 5 sequences and 30 sequences. We then compare the true sequence logos and EDLogo with those based on the PWM estimated by sample proportion and the PWM estimated by dash.</p>
<pre class="r"><code>get_viewport_logo(2,2)
seekViewport(paste0(&quot;plotlogo&quot;, 1))
set.seed(35)
pfms=dashsample(pfm,5)
logomaker(pfms,color_profile = color_profile,newpage = F,pop_name = &#39;Sample proportion,counts=5&#39;)
seekViewport(paste0(&quot;plotlogo&quot;, 2))
logomaker(getpwm(pfms,type = &#39;dash&#39;),color_profile = color_profile,newpage = F,pop_name = &#39;dash,counts=5&#39;)
seekViewport(paste0(&quot;plotlogo&quot;, 3))
pfms=dashsample(pfm,30)
logomaker(pfms,color_profile = color_profile,newpage = F,pop_name = &#39;Sample proportion,counts=30&#39;)
seekViewport(paste0(&quot;plotlogo&quot;, 4))
logomaker(getpwm(pfms,type = &#39;dash&#39;),color_profile = color_profile,newpage = F,pop_name = &#39;dash,counts=30&#39;)</code></pre>
<p><img src="figure/dashtutor2.Rmd/unnamed-chunk-3-1.png" width="864" style="display: block; margin: auto;" /></p>
<p>From the plot, we can see that when the counts are small, the shrinkage efects of dash are strong. The dash preserves the logos that have more information, e.g.Â position 4-8, while shrinks the heights of logos that do not show strong conservation. As the number of counts increase, the shrinkage efects are less strong while the sequence logos are more similar to the true plot.</p>
<p>This highlights why <em>dash</em> is an important tool to perform adaptive shrinkage of compositional data in case of low positional frequencies. The problem of low positional frequencies is highly relevant in Plant Transcription Factor Database <a href="http://planttfdb.cbi.pku.edu.cn/">PlantTFDB</a> and in several tranacription factor motifs in <a href="http://hocomoco11.autosome.ru/">HOCOMOCO</a>.</p>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<!-- Insert the session information into the document -->
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.4.2 (2017-09-28)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Sierra 10.12.5

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
[1] Logolas_1.2.1

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.13         knitr_1.16           magrittr_1.5        
 [4] munsell_0.4.3        gridBase_0.4-7       SQUAREM_2017.10-1   
 [7] colorspace_1.3-2     rlang_0.1.4          stringr_1.2.0       
[10] plyr_1.8.4           tools_3.4.2          parallel_3.4.2      
[13] gtable_0.2.0         git2r_0.19.0         htmltools_0.3.6     
[16] yaml_2.1.14          lazyeval_0.2.0       rprojroot_1.2       
[19] digest_0.6.12        tibble_1.3.4         RColorBrewer_1.1-2  
[22] ggplot2_2.2.1        evaluate_0.10.1      LaplacesDemon_16.0.1
[25] rmarkdown_1.6        stringi_1.1.5        compiler_3.4.2      
[28] scales_0.4.1         backports_1.1.0     </code></pre>
</div>

<hr>
<p>
    This webpage has been developed using RStudio's <a href="http://rmarkdown.rstudio.com">R Markdown</a> and John D Blischak's <a href="https://github.com/jdblischak/workflowr">workflowr</a> package. 
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
