<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="viewport" content="width=device-width, initial-scale=1">



<title>String Logos</title>

<link href="site_libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>




<link rel="stylesheet" href="cosmo.css" type="text/css" />

</head>

<body>


<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <link href="https://fonts.googleapis.com/css?family=Open+Sans:100,300,400,600" rel="stylesheet" type="text/css">
    <link href="../docs/cosmo.css" rel="stylesheet">
    <link href="../docs/cosmo.min.css" rel="stylesheet">
</head>
<script src="https://ajax.googleapis.com/ajax/libs/jquery/1.12.4/jquery.min.js"></script>
<script src="../docs/bootstrap/js/bootstrap.min.js"></script>

<body>

<div class = "nav" align = "middle">
<div class= "p1"></div>
<p>
  <a class="btn btn-large btn btn-primary" type="button" href = "Logolas_home.html">HOME</a>
  &nbsp &nbsp
  <a class="btn btn-large btn btn-primary" type="button" href = "Gallery.html">GALLERY</a>
  &nbsp &nbsp
  <a class="btn btn-large btn btn-primary" type="button" href = "https://github.com/kkdey/Logolas">SOFTWARE</a>
  &nbsp &nbsp
  <a class="btn btn-large btn btn-primary" type="button" href = "workflow.html">VIGNETTES</a>
  &nbsp &nbsp
  <a class="btn btn-large btn btn-primary" type="button" href = "license.html">LICENSE</a>
   &nbsp &nbsp
  <a class="btn btn-large btn btn-primary" type="button" href = "https://github.com/kkdey/Logolas-pages">GITHUB</a>
 </p>
</div>
<!-- <div class="nav">
  <a href="index.html">Home page</a> &nbsp &nbsp
  <a href="musings.html">Musings</a> &nbsp &nbsp
  <a href="town.html">My town</a> &nbsp &nbsp
  <a href="links.html">Links</a>
</div> -->

<!-- Main content -->
</body>
</html>


<h1 class="title toc-ignore">String Logos</h1>


<div id="TOC">
<ul>
<li><a href="#character-symbol-logos">Character Symbol Logos</a><ul>
<li><a href="#protein-example">Protein Example</a></li>
<li><a href="#pssm-plot">PSSM plot</a></li>
</ul></li>
<li><a href="#strings-character-symbols-logos">Strings + Character Symbols Logos</a><ul>
<li><a href="#mutation-signatures-plot">Mutation Signatures plot</a></li>
</ul></li>
<li><a href="#full-string-plots">Full string plots</a><ul>
<li><a href="#histone-mark-composition">Histone mark composition</a></li>
</ul></li>
<li><a href="#session-information">Session information</a></li>
</ul>
</div>

<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
<!-- Update knitr chunk options -->
<!-- Insert the date the file was last updated -->
<p><strong>Last updated:</strong> 2017-11-14</p>
<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
<p><strong>Code version:</strong> 35656d9</p>
<!-- Add your analysis here -->
<p>In this chapter, we highlight another key feature of <strong>Logolas</strong> - the ability to plot symbols not just for A, C, G and T, or English alphabets, but for any alphanumeric string.</p>
<div id="character-symbol-logos" class="section level2">
<h2>Character Symbol Logos</h2>
<div id="protein-example" class="section level3">
<h3>Protein Example</h3>
<p>We first present an example of visualizing protein binding sites using <strong>Logolas</strong> where each character is represented by an English alphabet. Keeping with the flexibility to use any background, for the protein data example here, the background <code>bg</code> is set to uniform in panel (a) and to the amino acid composition from BLOSUM62 in panel (b).</p>
<pre class="r"><code>library(Logolas)
library(grid)
counts_mat &lt;- rbind(c(0, 0, 100, 1, 2), c(4, 3, 30, 35, 2),
c(100, 0, 10, 2, 7),rep(0,5),
c(4, 2, 3, 7, 70), c(1, 8, 0, 60, 3),
rep(0, 5), c(4, 2, 100, 1, 1),
c(12, 8, 16, 7, 20), c(55, 0, 1, 0, 12),
rep(0,5), c(rep(0,3), 20, 0),
rep(0,5), c(0, 0, 30, 0, 22),
c(1, 0, 12, 3, 10), rep(0,5),
c(0, 1, 0, 34, 1), c(0, 1, 12, 35, 1),
c(0, 30, 1, 10, 2), c(0, 1, 4, 100, 2))

rownames(counts_mat) &lt;- c(&quot;A&quot;, &quot;R&quot;, &quot;N&quot;, &quot;D&quot;,&quot;C&quot;, &quot;E&quot;, &quot;Q&quot;, &quot;G&quot;,
&quot;H&quot;, &quot;I&quot;, &quot;L&quot;, &quot;K&quot;, &quot;M&quot;, &quot;F&quot;, &quot;P&quot;, &quot;S&quot;,
&quot;T&quot;, &quot;W&quot;, &quot;Y&quot;, &quot;V&quot;)
colnames(counts_mat) &lt;- c(&quot;Pos 1&quot;, &quot;Pos 2&quot;, &quot;Pos 3&quot;, &quot;Pos 4&quot;, &quot;Pos 5&quot;)
cols1 &lt;- c(rev(RColorBrewer::brewer.pal(12, &quot;Paired&quot;))[c(3,4,7,8,11,12,5,6,9,10)],
RColorBrewer::brewer.pal(12, &quot;Set3&quot;)[c(1,2,5,8,9)],
RColorBrewer::brewer.pal(9, &quot;Set1&quot;)[c(9,7)],
RColorBrewer::brewer.pal(8, &quot;Dark2&quot;)[c(3,4,8)])
color_profile &lt;- list(&quot;type&quot; = &quot;per_row&quot;,
&quot;col&quot; = cols1)


get_viewport_logo(1, 2)

seekViewport(paste0(&quot;plotlogo&quot;, 1))

logomaker(counts_mat,
color_profile = color_profile,
newpage = FALSE,
pop_name = &quot;Uniform Background&quot;,
frame_width = 1)

seekViewport(paste0(&quot;plotlogo&quot;, 2))

logomaker(counts_mat,
color_profile = color_profile,
bg=c(0.074,0.052,0.045,0.054,0.025,0.034,0.054,0.074,0.026,0.068,0.099,0.058,0.025,0.047,0.039,0.057,0.051,0.013,0.034,0.073),
newpage = FALSE,
pop_name = &#39;BLOSUM background&#39;,
frame_width = 1)</code></pre>
<p><img src="figure/extend.Rmd/unnamed-chunk-1-1.png" width="864" style="display: block; margin: auto;" /></p>
</div>
<div id="pssm-plot" class="section level3">
<h3>PSSM plot</h3>
<p>However for protein motif data, PWM is not always reported. Instead a derived score matrix called PSSM (Position specific scoring matrix) is often reported.</p>
<p>Logolas provides a function <code>logo_pssm</code> to plot the sequence logo based on the PSSM data. One can also apply median adjustment to the PSSM matrix like in  plot.</p>
<pre class="r"><code>get_viewport_logo(1, 2)
seekViewport(paste0(&quot;plotlogo&quot;, 1))
logo_pssm(pssm,
          color_profile = color_profile,
          control = list(quant = 0, gap_ylab = 3.5, round_off = 0),
          newpage = FALSE,
          pop_name = &quot;PSSM : no adj.&quot;)

seekViewport(paste0(&quot;plotlogo&quot;, 2))
logo_pssm(pssm,
          color_profile = color_profile,
          control = list(quant = 0.5, gap_ylab = 3.5, round_off = 0),
          newpage = FALSE,
          pop_name = &quot;PSSM: median adj.&quot;)</code></pre>
<p><img src="figure/extend.Rmd/unnamed-chunk-4-1.png" width="864" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="strings-character-symbols-logos" class="section level2">
<h2>Strings + Character Symbols Logos</h2>
<div id="mutation-signatures-plot" class="section level3">
<h3>Mutation Signatures plot</h3>
<p>Next we present an example of cancer mutation signature profiles where string symbols are mixed with character symbols.</p>
<p>A cancer mutation signature is usually represented by the type of mutation (C-&gt;T, C-&gt;A, C-&gt;G, T-&gt;A, T-&gt;C.T-&gt;G) along with the immediately surrounding left and right flanking bases, each of which can take the symbols A, C, G and T. We use the string symbol feature of <strong>Logolas</strong> to represent the substitutions <span class="math inline">\(X\rightarrow Y\)</span> and the character symbols for the flanking bases.</p>
<p>User may want to have the C in <span class="math inline">\(C&gt;T\)</span> to be of the same color as the C at the flanking bases and Logolas coloring profile provides the user the flexibility to color each symbol instead of a string. While <code>per_row</code> profile would assign a separate color to <span class="math inline">\(C-&gt;T\)</span> and the standalone base <span class="math inline">\(C\)</span> at the flanking bases, the <code>per_symbol</code> option can actually recognise each character in the string and assign a color to it. We see the difference through the example below.</p>
<pre class="r"><code>mFile &lt;- system.file(&quot;Exfiles/pwm1&quot;, package=&quot;seqLogo&quot;)
m &lt;- read.table(mFile)
p &lt;- seqLogo::makePWM(m)
pwm_mat &lt;- slot(p,name = &quot;pwm&quot;)
mat1 &lt;- cbind(pwm_mat[,c(3,4)], rep(NA,4), pwm_mat[,c(5,6)]);
colnames(mat1) &lt;- c(&quot;-2&quot;, &quot;-1&quot;, &quot;0&quot;, &quot;1&quot;, &quot;2&quot;)
mat2 &lt;- cbind(rep(NA,6), rep(NA,6),
c(0.5, 0.2, 0.2, 0.05, 0.05, 0),
rep(NA,6), rep(NA,6))
rownames(mat2) &lt;- c(&quot;C&gt;T&quot;, &quot;C&gt;A&quot;, &quot;C&gt;G&quot;,
&quot;T&gt;A&quot;, &quot;T&gt;C&quot;, &quot;T&gt;G&quot;)
table &lt;- rbind(mat1, mat2)

get_viewport_logo(1, 2)

seekViewport(paste0(&quot;plotlogo&quot;, 1))
color_profile &lt;- list(&quot;type&quot; = &quot;per_row&quot;,
&quot;col&quot; = RColorBrewer::brewer.pal(dim(table)[1],name =&quot;Spectral&quot;))
nlogomaker(table,
          color_profile = color_profile,
          logoheight = &quot;log&quot;,
          pop_name = &quot;color per element&quot;,
          newpage = FALSE)


seekViewport(paste0(&quot;plotlogo&quot;, 2))
cols = RColorBrewer::brewer.pal.info[RColorBrewer::brewer.pal.info$category == &#39;qual&#39;,]
col_vector = unlist(mapply(RColorBrewer::brewer.pal, cols$maxcolors, rownames(cols)))

total_chars = c(&quot;A&quot;, &quot;B&quot;, &quot;C&quot;, &quot;D&quot;, &quot;E&quot;, &quot;F&quot;, &quot;G&quot;, &quot;H&quot;, &quot;I&quot;, &quot;J&quot;, &quot;K&quot;, &quot;L&quot;, &quot;M&quot;, &quot;N&quot;, &quot;O&quot;,
&quot;P&quot;, &quot;Q&quot;, &quot;R&quot;, &quot;S&quot;, &quot;T&quot;, &quot;U&quot;, &quot;V&quot;, &quot;W&quot;, &quot;X&quot;, &quot;Y&quot;, &quot;Z&quot;, &quot;zero&quot;, &quot;one&quot;, &quot;two&quot;,
&quot;three&quot;, &quot;four&quot;, &quot;five&quot;, &quot;six&quot;, &quot;seven&quot;, &quot;eight&quot;, &quot;nine&quot;, &quot;dot&quot;, &quot;comma&quot;,
&quot;dash&quot;, &quot;colon&quot;, &quot;semicolon&quot;, &quot;leftarrow&quot;, &quot;rightarrow&quot;)

set.seed(20)
color_profile &lt;- list(&quot;type&quot; = &quot;per_symbol&quot;,
&quot;col&quot; = sample(col_vector, length(total_chars), replace=FALSE))

nlogomaker(table,
          color_profile = color_profile,
          logoheight = &quot;log&quot;,
          total_chars = total_chars,
          pop_name = &quot;color per character&quot;,
          newpage = FALSE)</code></pre>
<p><img src="figure/extend.Rmd/unnamed-chunk-5-1.png" width="864" style="display: block; margin: auto;" /></p>
<p>Another coloring option available in <strong>Logolas</strong> is <code>per_column</code>, in which we assign a specific color for a specific column. This sort of coloring might be useful when the user wants to highlight the difference between columns [Check Gallery example of Birds data]</p>
</div>
</div>
<div id="full-string-plots" class="section level2">
<h2>Full string plots</h2>
<p>Next we consider an example where all the symbols are alphanumeric strings.</p>
<div id="histone-mark-composition" class="section level3">
<h3>Histone mark composition</h3>
<p>We apply Logolas on an example data from <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1891331/">Koch et al (2007)</a> that summarizes the relative abundance patterns 5 different histone marks in different genomic contexts - intergenic sequence, intron, exon, gene start and gene end for the lymphoblastoid cell line, GM06990, in the ChIP-CHIP data. Note here that the histone marks are alphanumeric symbols, for example, H3K4ME1. we use as background a table of relative abundances generated from randomly simulated data in Koch <em>et al</em> (2007).</p>
<pre class="r"><code>mat &lt;- rbind(c(326, 296, 81, 245, 71),
             c(258, 228, 55, 273, 90),
             c(145, 121, 29, 253, 85),
             c(60, 52, 23, 180, 53),
             c(150, 191, 63, 178, 63))

bgmat &lt;- rbind(c(542, 218, 118, 108, 33),
               c(480, 204, 107, 87, 26),
               c(346, 131, 71, 66, 20),
               c(199, 72, 41, 43, 13),
               c(368, 101, 67, 80, 30))

rownames(mat) &lt;- c(&quot;H3K4ME1&quot;, &quot;H3K4ME2&quot;, &quot;H3K4ME3&quot;, &quot;H3AC&quot;, &quot;H4AC&quot;)
colnames(mat) &lt;- c(&quot;Intergenic&quot;,&quot;Intron&quot;,&quot;Exon&quot;,
                   &quot;Gene start &quot;,&quot;Gene end&quot;)

set.seed(201)
color_profile &lt;- list(&quot;type&quot; = &quot;per_row&quot;,
                      &quot;col&quot; = sample(RColorBrewer::brewer.pal(10,name = &quot;Spectral&quot;), dim(mat)[1]))

get_viewport_logo(1, 2)

seekViewport(paste0(&quot;plotlogo&quot;, 1))
Logolas::logomaker(mat,
          color_profile = color_profile,
          pop_name = &quot;&quot;,
          newpage = FALSE,
          col_line_split = &quot;black&quot;,
          control = list(gap_ylab = 3.5,
                         totbins = 3))

seekViewport(paste0(&quot;plotlogo&quot;, 2))
Logolas::nlogomaker(mat,
                    logoheight = &#39;log&#39;,
                    bg = bgmat,
                    pop_name = &quot;&quot;,
                    color_profile = color_profile,
                    newpage = FALSE,
                    control = list(gap_ylab=3.5,
                                   round_off = 1,
                                   posbins = 3, negbins = 3))</code></pre>
<p><img src="figure/extend.Rmd/unnamed-chunk-6-1.png" width="864" style="display: block; margin: auto;" /></p>
<p>The control options <code>posbins</code>, <code>negbins</code> and <code>totbins</code> determine the number of ticks in the positive and negative parts of the Y axis in the logo plot, whereas <code>gap_ylab</code> denotes the spacing between the Y axis label and the y axis ticks.</p>
</div>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<!-- Insert the session information into the document -->
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.4.2 (2017-09-28)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Sierra 10.12.5

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
[1] Logolas_1.2.1

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.13       knitr_1.16         magrittr_1.5      
 [4] munsell_0.4.3      gridBase_0.4-7     SQUAREM_2017.10-1 
 [7] colorspace_1.3-2   rlang_0.1.4        stringr_1.2.0     
[10] plyr_1.8.4         tools_3.4.2        gtable_0.2.0      
[13] seqLogo_1.44.0     git2r_0.19.0       htmltools_0.3.6   
[16] yaml_2.1.14        lazyeval_0.2.0     rprojroot_1.2     
[19] digest_0.6.12      tibble_1.3.4       RColorBrewer_1.1-2
[22] ggplot2_2.2.1      evaluate_0.10.1    rmarkdown_1.6     
[25] stringi_1.1.5      compiler_3.4.2     scales_0.4.1      
[28] backports_1.1.0    stats4_3.4.2      </code></pre>
</div>

<hr>
<p>
    This webpage has been developed using RStudio's <a href="http://rmarkdown.rstudio.com">R Markdown</a> and John D Blischak's <a href="https://github.com/jdblischak/workflowr">workflowr</a> package. 
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
