<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="viewport" content="width=device-width, initial-scale=1">



<title>Extended features of Logolas</title>

<link href="site_libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>




<link rel="stylesheet" href="cosmo.css" type="text/css" />

</head>

<body>


<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <link href="https://fonts.googleapis.com/css?family=Open+Sans:100,300,400,600" rel="stylesheet" type="text/css">
    <link href="../docs/cosmo.css" rel="stylesheet">
    <link href="../docs/cosmo.min.css" rel="stylesheet">
</head>
<script src="https://ajax.googleapis.com/ajax/libs/jquery/1.12.4/jquery.min.js"></script>
<script src="../docs/bootstrap/js/bootstrap.min.js"></script>

<body>

<div class = "nav" align = "middle">
<div class= "p1"></div>
<p>
  <a class="btn btn-large btn btn-primary" type="button" href = "Logolas_home.html">HOME</a>
  &nbsp &nbsp
  <a class="btn btn-large btn btn-primary" type="button" href = "Gallery.html">GALLERY</a>
  &nbsp &nbsp
  <a class="btn btn-large btn btn-primary" type="button" href = "https://github.com/kkdey/Logolas">SOFTWARE</a>
  &nbsp &nbsp
  <a class="btn btn-large btn btn-primary" type="button" href = "workflow.html">VIGNETTES</a>
  &nbsp &nbsp
  <a class="btn btn-large btn btn-primary" type="button" href = "license.html">LICENSE</a>
   &nbsp &nbsp
  <a class="btn btn-large btn btn-primary" type="button" href = "https://github.com/kkdey/Logolas-pages">GITHUB</a>
 </p>
</div>
<!-- <div class="nav">
  <a href="index.html">Home page</a> &nbsp &nbsp
  <a href="musings.html">Musings</a> &nbsp &nbsp
  <a href="town.html">My town</a> &nbsp &nbsp
  <a href="links.html">Links</a>
</div> -->

<!-- Main content -->
</body>
</html>


<h1 class="title toc-ignore">Extended features of Logolas</h1>


<div id="TOC">
<ul>
<li><a href="#protein-logo-plot">Protein logo plot</a></li>
<li><a href="#pssm-plot">PSSM plot</a></li>
<li><a href="#mutation-signature-plot">Mutation signature plot</a></li>
<li><a href="#full-string-plots">Full string plots</a><ul>
<li><a href="#histone-marks">Histone marks</a></li>
<li><a href="#document-mining">Document mining</a></li>
</ul></li>
<li><a href="#combining-plots-with-r-graphs">Combining plots with R graphs</a></li>
<li><a href="#session-information">Session information</a></li>
</ul>
</div>

<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
<!-- Update knitr chunk options -->
<!-- Insert the date the file was last updated -->
<p><strong>Last updated:</strong> 2017-09-15</p>
<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
<p><strong>Code version:</strong> 0de44dc</p>
<!-- Add your analysis here -->
<div id="protein-logo-plot" class="section level2">
<h2>Protein logo plot</h2>
<p>User could use Logolas for protein sequence motif plot as well, as its logo library includes all the English alphabets and the 20 amino acids have a 1-letter representation using English alphabets.</p>
<p>Note that all one needs to do to build the logo plots is to specify the row names and column names as per the the logos and the stack labels and then fix the colors for the logos.</p>
<p>The default background probability is uniform and user could specify the background probability for each amino acid. One common used is from the BLOSUM62.</p>
<pre class="r"><code>library(Logolas)
library(grid)
counts_mat &lt;- rbind(c(0, 0, 100, 1, 2), c(4, 3, 30, 35, 2),
c(100, 0, 10, 2, 7),rep(0,5),
c(4, 2, 3, 7, 70), c(1, 8, 0, 60, 3),
rep(0, 5), c(4, 2, 100, 1, 1),
c(12, 8, 16, 7, 20), c(55, 0, 1, 0, 12),
rep(0,5), c(rep(0,3), 20, 0),
rep(0,5), c(0, 0, 30, 0, 22),
c(1, 0, 12, 3, 10), rep(0,5),
c(0, 1, 0, 34, 1), c(0, 1, 12, 35, 1),
c(0, 30, 1, 10, 2), c(0, 1, 4, 100, 2))

rownames(counts_mat) &lt;- c(&quot;A&quot;, &quot;R&quot;, &quot;N&quot;, &quot;D&quot;,&quot;C&quot;, &quot;E&quot;, &quot;Q&quot;, &quot;G&quot;,
&quot;H&quot;, &quot;I&quot;, &quot;L&quot;, &quot;K&quot;, &quot;M&quot;, &quot;F&quot;, &quot;P&quot;, &quot;S&quot;,
&quot;T&quot;, &quot;W&quot;, &quot;Y&quot;, &quot;V&quot;)
colnames(counts_mat) &lt;- c(&quot;Pos 1&quot;, &quot;Pos 2&quot;, &quot;Pos 3&quot;, &quot;Pos 4&quot;, &quot;Pos 5&quot;)
cols1 &lt;- c(rev(RColorBrewer::brewer.pal(12, &quot;Paired&quot;))[c(3,4,7,8,11,12,5,6,9,10)],
RColorBrewer::brewer.pal(12, &quot;Set3&quot;)[c(1,2,5,8,9)],
RColorBrewer::brewer.pal(9, &quot;Set1&quot;)[c(9,7)],
RColorBrewer::brewer.pal(8, &quot;Dark2&quot;)[c(3,4,8)])
color_profile &lt;- list(&quot;type&quot; = &quot;per_row&quot;,
&quot;col&quot; = cols1)

grid.newpage()
layout.rows &lt;- 1
layout.cols &lt;- 2
top.vp &lt;- viewport(layout=grid.layout(layout.rows,layout.cols, widths=unit(rep(2,layout.cols), rep(&quot;null&quot;,layout.cols)),heights=unit(rep(2,1), rep(&quot;null&quot;,1))))

plot_reg &lt;- vpList()
l &lt;- 1
for(i in 1:layout.rows){
  for(j in 1:layout.cols){
    plot_reg[[l]] &lt;- viewport(layout.pos.col = j, layout.pos.row = i, name = paste0(&quot;plotlogo&quot;, l))
    l &lt;- l+1
  }
}


plot_tree &lt;- vpTree(top.vp, plot_reg)

pushViewport(plot_tree)
seekViewport(paste0(&quot;plotlogo&quot;, 1))

logomaker(counts_mat,
color_profile = color_profile,
newpage = FALSE,
frame_width = 1)

seekViewport(paste0(&quot;plotlogo&quot;, 2))

logomaker(counts_mat,
color_profile = color_profile,
bg=c(0.074,0.052,0.045,0.054,0.025,0.034,0.054,0.074,0.026,0.068,0.099,0.058,0.025,0.047,0.039,0.057,0.051,0.013,0.034,0.073),
newpage = FALSE,
pop_name = &#39;Logo plot with Varying background prob&#39;,
frame_width = 1)</code></pre>
<p><img src="figure/extend.Rmd/unnamed-chunk-1-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="pssm-plot" class="section level2">
<h2>PSSM plot</h2>
<p>Position-specific scoring matrix(PSSM) is constructed from a multiple alignment of a set of sequences. If a residue at a particular position is highly conserved(high relative frequency), it is given a high positive socre and the other residues are given high negative score. All residues are given scores near zero if the residues are all weakly conserved at that position.</p>
<p>The function <code>logo_pssm</code> plots the sequence logo based on the PSSM.</p>
<pre class="r"><code>pssm</code></pre>
<pre><code>   1  2  3  4  5  6  7  8  9
A -2 -2 -3 -3 -1  0 -4 -3 -1
R  1 -4  0 -1 -2 -2 -1  1  4
N  2  1 -4 -5  3  3 -4 -5 -2
D -2  0 -4 -5 -2  6 -5 -5 -2
C -4 -4 -3 -3 -4 -5 -4 -3 -5
Q  3  0  1 -3  1  2 -4 -3  1
E  2 -3 -3 -2  3  3 -2 -4  2
G -4  6 -5 -5 -1 -3 -5 -5 -3
H  2 -4 -4 -4  2  1 -2 -4 -2
I -3 -2  1 -1 -3 -5 -2 -1 -5
L -2 -5  4  5  0 -5  0  5 -4
K  4 -3  1 -4  2 -1 -4 -3  6
M -1 -4  3  0  1 -4 -2  1 -3
F -4 -5 -2 -1 -4 -5  7 -2 -5
P -3 -4 -4  0 -3 -3 -5 -5 -3
S  0 -1 -2 -2  0 -1 -4 -2 -2
T -1 -3  1 -3  0 -1 -4 -3 -2
W -4 -5 -4  7 -5 -5 -1  5 -5
Y -1 -5 -3  1 -3 -4  5 -2 -4
V  0 -4  1 -2 -1 -4 -1  0 -4</code></pre>
<pre class="r"><code>logo_pssm(pssm,
          color_profile = color_profile,
          frame_width = 1
          )</code></pre>
<p><img src="figure/extend.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="mutation-signature-plot" class="section level2">
<h2>Mutation signature plot</h2>
<p>Logolas could plot strings as logos. The first example is the cancer mutation signature. Detection of cancer mutaiton signatures is important in the study of the causes and mechanism of cancers. Usually we consider 6 possible mutation patterns(C to A/G/T, T to A/C/G). Suppose for a set of cell lines, we are provided data on the number of mutations (nucleotide substitions) and nucleotides at the flanking bases of the nucleotide substitution.</p>
<p>Shiraishi <em>et al.</em>(2015) proposed a parsimonious approach to model the mutation signatures, by assuming independence across mutation patterns. We apply it here on a demo example.</p>
<p>we use the symbols <span class="math inline">\(X&gt;Y\)</span> to denote the <span class="math inline">\(X\rightarrow Y\)</span> substitutions. The data contains proportion of logos in each position : left flanking, left flanking, mutation, 1 right flanking and 2 right flanking. Note that <span class="math inline">\(X&gt;Y\)</span> type symbols occur only in the middle stack (column) as that is the mutation stack, while the nucleotides A, C, T and G occur only in the left two and right two flanking bases stacks (columns).</p>
<p>User may want to have the C in <span class="math inline">\(C&gt;T\)</span> to be of the same color as the C at the flankings and Logolas coloring profile provides the user the flexibility to color each symbol instead of a string. We use the color type <code>per_symbol</code> instead of the <code>per_row</code> profile we have been using so far. We consider a list of all symbols <code>total_chars</code> which is set as default to the list chosen above (so you can skip the <code>total_chars</code> argument below). However, if the user adds a symbol to the library (the process of doing that we show in the end), then the library is expected to grow and the user then might want to update the total_chars list by adding new symbols.</p>
<p>Another coloring option is <code>per_column</code>, in which we have a specific color for a specific column. This sort of coloring might be useful when the user wants to highlight the difference between columns. We provide an example of this next.</p>
<pre class="r"><code>mFile &lt;- system.file(&quot;Exfiles/pwm1&quot;, package=&quot;seqLogo&quot;)
m &lt;- read.table(mFile)
p &lt;- seqLogo::makePWM(m)
pwm_mat &lt;- slot(p,name = &quot;pwm&quot;)
mat1 &lt;- cbind(pwm_mat[,c(3,4)], rep(NA,4), pwm_mat[,c(5,6)]);
colnames(mat1) &lt;- c(&quot;-2&quot;, &quot;-1&quot;, &quot;0&quot;, &quot;1&quot;, &quot;2&quot;)
mat2 &lt;- cbind(rep(NA,6), rep(NA,6),
c(0.5, 0.2, 0.2, 0.05, 0.05, 0),
rep(NA,6), rep(NA,6))
rownames(mat2) &lt;- c(&quot;C&gt;T&quot;, &quot;C&gt;A&quot;, &quot;C&gt;G&quot;,
&quot;T&gt;A&quot;, &quot;T&gt;C&quot;, &quot;T&gt;G&quot;)
table &lt;- rbind(mat1, mat2)

library(grid)
grid.newpage()
layout.rows &lt;- 1
layout.cols &lt;- 2
top.vp &lt;- viewport(layout=grid.layout(layout.rows, layout.cols,
                                      widths=unit(rep(6,layout.cols), rep(&quot;null&quot;, 2)),
                                      heights=unit(c(20,50), rep(&quot;lines&quot;, 2))))

plot_reg &lt;- vpList()
l &lt;- 1
for(i in 1:layout.rows){
  for(j in 1:layout.cols){
    plot_reg[[l]] &lt;- viewport(layout.pos.col = j, layout.pos.row = i, name = paste0(&quot;plotlogo&quot;, l))
    l &lt;- l+1
  }
}


plot_tree &lt;- vpTree(top.vp, plot_reg)

pushViewport(plot_tree)
seekViewport(paste0(&quot;plotlogo&quot;, 1))
color_profile &lt;- list(&quot;type&quot; = &quot;per_row&quot;,
&quot;col&quot; = RColorBrewer::brewer.pal(dim(table)[1],name =&quot;Spectral&quot;))
logomaker(table,
          color_profile = color_profile,
          frame_width = 1,
          newpage = FALSE,
          xlab = &quot;Position&quot;,
          ylab = &quot;Information content&quot;)


seekViewport(paste0(&quot;plotlogo&quot;, 2))
cols = RColorBrewer::brewer.pal.info[RColorBrewer::brewer.pal.info$category == &#39;qual&#39;,]
col_vector = unlist(mapply(RColorBrewer::brewer.pal, cols$maxcolors, rownames(cols)))

total_chars = c(&quot;A&quot;, &quot;B&quot;, &quot;C&quot;, &quot;D&quot;, &quot;E&quot;, &quot;F&quot;, &quot;G&quot;, &quot;H&quot;, &quot;I&quot;, &quot;J&quot;, &quot;K&quot;, &quot;L&quot;, &quot;M&quot;, &quot;N&quot;, &quot;O&quot;,
&quot;P&quot;, &quot;Q&quot;, &quot;R&quot;, &quot;S&quot;, &quot;T&quot;, &quot;U&quot;, &quot;V&quot;, &quot;W&quot;, &quot;X&quot;, &quot;Y&quot;, &quot;Z&quot;, &quot;zero&quot;, &quot;one&quot;, &quot;two&quot;,
&quot;three&quot;, &quot;four&quot;, &quot;five&quot;, &quot;six&quot;, &quot;seven&quot;, &quot;eight&quot;, &quot;nine&quot;, &quot;dot&quot;, &quot;comma&quot;,
&quot;dash&quot;, &quot;colon&quot;, &quot;semicolon&quot;, &quot;leftarrow&quot;, &quot;rightarrow&quot;)

set.seed(20)
color_profile &lt;- list(&quot;type&quot; = &quot;per_symbol&quot;,
&quot;col&quot; = sample(col_vector, length(total_chars), replace=FALSE))

logomaker(table,
          color_profile = color_profile,
          total_chars = total_chars,
          frame_width = 1,
          newpage = FALSE,
          xlab = &quot;Position&quot;,
          ylab = &quot;Information content&quot;)</code></pre>
<p><img src="figure/extend.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="full-string-plots" class="section level2">
<h2>Full string plots</h2>
<p>Logolas provides a handy visualization of strings, apart from the simple alphabet. We present two examples below.</p>
<div id="histone-marks" class="section level3">
<h3>Histone marks</h3>
<p>In studies related to histone marks, one might be interested to see if certain histone marks are prominent than others in some cell lines or tissues or in some genomic regions. In this case, we apply Logolas on an example data from Koch <em>et al</em> (2007). The authors recorded number of histone modification sites identified by their algorithm which overlap with an intergenic sequence, intron, exon, gene start and gene end for the lymphoblastoid cell line, GM06990, in the ChIP-CHIP data. Note here that the histone mark symbols are alphanumeric, for example H3K4ME1.</p>
<pre class="r"><code>mat &lt;- rbind(c(326, 296, 81, 245, 71),
c(258, 228, 55, 273, 90),
c(145, 121, 29, 253, 85),
c(60, 52, 23, 180, 53),
c(150, 191, 63, 178, 63))
rownames(mat) &lt;- c(&quot;H3K4ME1&quot;, &quot;H3K4ME2&quot;, &quot;H3K4ME3&quot;, &quot;H3AC&quot;, &quot;H4AC&quot;)
colnames(mat) &lt;- c(&quot;Intergenic&quot;,&quot;Intron&quot;,&quot;Exon \n 1000 KB window&quot;,
&quot;Gene start \n 1000 KB window&quot;,&quot;Gene end \n 1000 KB window&quot;)

color_profile &lt;- list(&quot;type&quot; = &quot;per_row&quot;,
&quot;col&quot; = RColorBrewer::brewer.pal(dim(mat)[1],name =&quot;Spectral&quot;))

logomaker(mat,color_profile = color_profile,frame_width = 1, ic.scale = TRUE, yscale_change = TRUE)</code></pre>
<p><img src="figure/extend.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>From the plot, it’s practical to see how the patterns of histone modification sites changes across genomic region types for that cell line.</p>
</div>
<div id="document-mining" class="section level3">
<h3>Document mining</h3>
<p>Further, the logos could include the punctuation marks as well like the full stop.</p>
<p>Here we build a logo plot of the field categories of manuscipts submitted on aRxiv by 4 professors from Department of Statistics, University of Chicago. Note that the field categories here are a combination of numbers, alphabets, dots and dashes.</p>
<pre class="r"><code>library(aRxiv)
rec1 &lt;- aRxiv::arxiv_search(&#39;au:&quot;Matthew Stephens&quot;&#39;, limit=50)
rec2 &lt;- aRxiv::arxiv_search(&#39;au:&quot;John Lafferty&quot;&#39;, limit=50)
rec3 &lt;- aRxiv::arxiv_search(&#39;au:&quot;Wei Biao Wu&quot;&#39;, limit=50)
rec4 &lt;- aRxiv::arxiv_search(&#39;au:&quot;Peter Mccullagh&quot;&#39;, limit=50)

primary_categories_1 &lt;- toupper(rec1$primary_category)
primary_categories_2 &lt;- toupper(rec2$primary_category)
primary_categories_3 &lt;- toupper(rec3$primary_category)
primary_categories_4 &lt;- toupper(rec4$primary_category)

factor_levels &lt;- unique(c(unique(primary_categories_1),
unique(primary_categories_2),
unique(primary_categories_3),
unique(primary_categories_4)))

primary_categories_1 &lt;- factor(primary_categories_1, levels=factor_levels)
primary_categories_2 &lt;- factor(primary_categories_2, levels=factor_levels)
primary_categories_3 &lt;- factor(primary_categories_3, levels=factor_levels)
primary_categories_4 &lt;- factor(primary_categories_4, levels=factor_levels)

tab_data &lt;- cbind(table(primary_categories_1),
table(primary_categories_2),
table(primary_categories_3),
table(primary_categories_4))

colnames(tab_data) &lt;- c(&quot;Matthew Stephens&quot;,
&quot;John Lafferty&quot;,
&quot;Wei Biao Wu&quot;,
&quot;Peter McCullagh&quot;)
tab_data &lt;- as.matrix(tab_data)

color_profile &lt;- list(&quot;type&quot; = &quot;per_row&quot;,
&quot;col&quot; = RColorBrewer::brewer.pal(dim(tab_data)[1],
name = &quot;Spectral&quot;))

logomaker(tab_data,
        color_profile = color_profile,
        frame_width = 1,
        pop_name = &quot;arXiv field categories of UChicago STAT professors&quot;,
        xlab = &quot;Professors&quot;,
        ylab = &quot;Information content&quot;)</code></pre>
<p><img src="figure/extend.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="combining-plots-with-r-graphs" class="section level2">
<h2>Combining plots with R graphs</h2>
<p>User could combine logo plots from Logolas with plots from the other R graphs such as R base graphics and ggplot2 graph.</p>
<pre class="r"><code>library(ggplot2)
library(grid)
library(gridBase)

m = matrix(rep(0,48),4,12)
m[1,] = c(0,0,2.5,7,0,0,0,0,0,0,1,2)
m[2,] = c(4,6,3,1,0,0,0,0,0,5,0,2)
m[3,] = c(0,0,0,0,0,1,8,0,0,1,1,2)
m[4,] = c(4,2,2.5,0,8,7,0,8,8,2,6,2)
rownames(m) = c(&quot;A&quot;, &quot;C&quot;, &quot;G&quot;, &quot;T&quot;)
colnames(m) = 1:12
m=m/8

color_profile = list(&quot;type&quot; = &quot;per_row&quot;,
                     &quot;col&quot; = RColorBrewer::brewer.pal(4,name =&quot;Spectral&quot;))

mtcars$gear &lt;- factor(mtcars$gear,levels=c(3,4,5),
                      labels=c(&quot;3gears&quot;,&quot;4gears&quot;,&quot;5gears&quot;))
mtcars$am &lt;- factor(mtcars$am,levels=c(0,1),
                    labels=c(&quot;Automatic&quot;,&quot;Manual&quot;))
mtcars$cyl &lt;- factor(mtcars$cyl,levels=c(4,6,8),
                     labels=c(&quot;4cyl&quot;,&quot;6cyl&quot;,&quot;8cyl&quot;))


grid.newpage()
layout.rows &lt;- 1
layout.cols &lt;- 2
top.vp &lt;- viewport(layout=grid.layout(layout.rows, layout.cols,
                                      widths=unit(rep(2,layout.cols), rep(&quot;null&quot;, layout.cols)),
                                      heights=unit(rep(2,1), rep(&quot;null&quot;,1))))

plot_reg &lt;- vpList()
l &lt;- 1
for(i in 1:layout.rows){
  for(j in 1:layout.cols){
    plot_reg[[l]] &lt;- viewport(layout.pos.col = j, layout.pos.row = i, name = paste0(&quot;plotlogo&quot;, l))
    l &lt;- l+1
  }
}


plot_tree &lt;- vpTree(top.vp, plot_reg)

pushViewport(plot_tree)
seekViewport(paste0(&quot;plotlogo&quot;, 1))
logomaker(m,xlab = &#39;position&#39;,color_profile = color_profile,
          bg = c(0.28, 0.22, 0.24, 0.26),
          frame_width = 1,
          newpage = FALSE,
          control = list(viewport.margin.left = 5))


seekViewport(paste0(&quot;plotlogo&quot;, 2))
vp3 = viewport(width=0.8, height=0.8, x = 0.5, y = 0.5)
p &lt;- qplot(mpg, data=mtcars, geom=&quot;density&quot;)
print(p, vp = vp3)</code></pre>
<p><img src="figure/extend.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># seekViewport(paste0(&quot;plotlogo&quot;, 3))
# vp3 = viewport(height = 1, width = 1)
# pushViewport(vp3)
# par(fig=gridFIG())
# par(gridPAR())
# #par(new=TRUE)
# #par(mar = c(3, 3, 1, 1))
# plot(1:10, type=&quot;b&quot;)</code></pre>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<!-- Insert the session information into the document -->
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.3.3 (2017-03-06)
Platform: x86_64-apple-darwin13.4.0 (64-bit)
Running under: macOS Sierra 10.12.5

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
[1] gridBase_0.4-7   ggplot2_2.2.1    aRxiv_0.5.15     dash_0.99.0     
[5] SQUAREM_2016.8-2 Logolas_1.1.2   

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.12         RColorBrewer_1.1-2   git2r_0.18.0        
 [4] plyr_1.8.4           tools_3.3.3          digest_0.6.12       
 [7] evaluate_0.10        tibble_1.3.4         gtable_0.2.0        
[10] rlang_0.1.1.9000     curl_2.6             yaml_2.1.14         
[13] parallel_3.3.3       LaplacesDemon_16.0.1 seqLogo_1.40.0      
[16] httr_1.2.1           stringr_1.2.0        knitr_1.16          
[19] stats4_3.3.3         rprojroot_1.2        R6_2.2.2            
[22] XML_3.98-1.9         rmarkdown_1.6        magrittr_1.5        
[25] backports_1.0.5      scales_0.4.1         htmltools_0.3.6     
[28] colorspace_1.3-2     labeling_0.3         stringi_1.1.5       
[31] lazyeval_0.2.0       munsell_0.4.3       </code></pre>
</div>

<hr>
<p>
    This webpage has been developed using RStudio's <a href="http://rmarkdown.rstudio.com">R Markdown</a> and John D Blischak's <a href="https://github.com/jdblischak/workflowr">workflowr</a> package. 
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
