<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="viewport" content="width=device-width, initial-scale=1">

<meta name="author" content="Dongyue Xie" />

<meta name="date" content="2017-08-23" />

<title>Introduction</title>

<link href="site_libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>




<link rel="stylesheet" href="cosmo.css" type="text/css" />

</head>

<body>


<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <link href="https://fonts.googleapis.com/css?family=Open+Sans:100,300,400,600" rel="stylesheet" type="text/css">
    <link href="../docs/cosmo.css" rel="stylesheet">
    <link href="../docs/cosmo.min.css" rel="stylesheet">
</head>
<script src="https://ajax.googleapis.com/ajax/libs/jquery/1.12.4/jquery.min.js"></script>
<script src="../docs/bootstrap/js/bootstrap.min.js"></script>

<body>

<div class = "nav" align = "middle">
<div class= "p1"></div>
<p>
  <a class="btn btn-large btn btn-primary" type="button" href = "Logolas_home.html">HOME</a> 
  &nbsp &nbsp
  <a class="btn btn-large btn btn-primary" type="button" href = "Gallery.html">GALLERY</a> 
  &nbsp &nbsp
  <a class="btn btn-large btn btn-primary" type="button" href = "https://github.com/kkdey/Logolas">SOFTWARE</a> 
  &nbsp &nbsp
  <a class="btn btn-large btn btn-primary" type="button" href = "https://dongyuexie.github.io/Sequence-Motif-Depletion/">VIGNETTES</a> 
  &nbsp &nbsp
  <a class="btn btn-large btn btn-primary" type="button" href = "license.html">LICENSE</a> 
   &nbsp &nbsp
  <a class="btn btn-large btn btn-primary" type="button" href = "https://github.com/kkdey/Logolas-pages">GITHUB</a> 
 </p>
</div>  
<!-- <div class="nav">
  <a href="index.html">Home page</a> &nbsp &nbsp
  <a href="musings.html">Musings</a> &nbsp &nbsp
  <a href="town.html">My town</a> &nbsp &nbsp
  <a href="links.html">Links</a>
</div> -->

<!-- Main content -->
</body>
</html>


<h1 class="title toc-ignore">Introduction</h1>
<h4 class="author"><em>Dongyue Xie</em></h4>
<h4 class="date"><em>2017-08-23</em></h4>


<div id="TOC">
<ul>
<li><a href="#reference">Reference</a><ul>
<li><a href="#session-information">Session information</a></li>
</ul></li>
</ul>
</div>

<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
<!-- Update knitr chunk options -->
<!-- Insert the date the file was last updated -->
<p><strong>Last updated:</strong> 2017-09-13</p>
<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
<p><strong>Code version:</strong> 485cb2a</p>
<!-- Add your analysis here -->
<p>Sequence logos is a graphical representation of patterns in the aligned sets of sequences. It was first introduced by Schneider&amp;Stephens(1990). Sequence logos is an informative visualization of patterns and easily shows the residue preferences and conservation of a particular position. These patterns are suggestive, in the sense of the key functional sites of transcription factor or protein family.</p>
<p>There are many existing packages or tools for generating sequence logos. For example, the R package <code>seqLogo</code>for displaying DNA sequence motif, web-based tool <code>WebLogo</code> , and <code>Seq2Logo</code> for construction and visualization of amino acid binding motifs.</p>
<p>However, certain limitations exist in these packages or tools. In particular, they have limited size of symbol library that can be used to make the plot, which greatly constrain the scope of applications. For example, the <code>seqLogo</code> could only plot logos of alphabet.</p>
<p>To address these constraints and make various applications possible, we introduce a new R package <code>Logolas</code>, for displaying logos that are not only standard but also user-specified. The logos could be alphabet, numbers, punctuation and etc. What is more important is that Logolas lets the user plot alphanumeric strings as logos, which extends the scope of the visualization massively, beyond the TFs or protein sequences. We will show examples of how this string representation of logos is effective in visualizing mutation signature patterns, ecological species abundance patterns etc.</p>
<p>Besides, the common sequence logos could only show the enrichment of residues, while the depletion is usually overlooked. In Logolas, we develop the negative logo plot that could display both the enrichment and depletion at one time. This would provide more insights and information.</p>
<p>Furthermore, based on the enrichment and depletion, a new symbol-calling way is proposed. This Logolas based nomenclature is a more generic alternative to the IUPAC and the Prosite nomenclatures used for calling nucleotide and amino acid respectively.</p>
<p>Most existing logo plotting tools take position weight matrix as input for making the logo plots. But this approach neglects the frequency scale underlying the position weight matrix. For example, a position weight computed for TF data based on just 10 fragments mapping to that position is less reliable compared to one based on 100 fragments. In such a case, the user would want to shrink the composition probability to the pre-defined background or prior much more in the first case compared to the second.</p>
<p>In Logolas, we provide a Dirichlet Adaptive Shrinkage method (dash), in similar lines to the adaptive shrinkage(ash) approach due to Stephens(2016), to adaptively shrink the positional weights based on the positional frequency scale.</p>
<p>Besides all the above benefits, Logolas allows for new and flexible stylizations, textures and color patterns of the logos , allows the user to create her own logos and add them to the library, and also provides an easy interface to combine Logolas plots with R base graphics and ggplot2 graphs in multi-panel visualizations.</p>
<div id="reference" class="section level1">
<h1>Reference</h1>
<p>Schneider, T. D., &amp; Stephens, R. M. (1990). Sequence logos: a new way to display consensus sequences. Nucleic acids research, 18(20), 6097-6100.</p>
<p>Stephens, M. (2016). False discovery rates: a new deal. Biostatistics, 18(2), 275-294.</p>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<!-- Insert the session information into the document -->
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.3.3 (2017-03-06)
Platform: x86_64-apple-darwin13.4.0 (64-bit)
Running under: macOS Sierra 10.12.5

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
 [1] backports_1.0.5 magrittr_1.5    rprojroot_1.2   tools_3.3.3    
 [5] htmltools_0.3.6 yaml_2.1.14     Rcpp_0.12.12    stringi_1.1.5  
 [9] rmarkdown_1.6   knitr_1.16      git2r_0.18.0    stringr_1.2.0  
[13] digest_0.6.12   evaluate_0.10  </code></pre>
</div>
</div>

<hr>
<p>
    This webpage has been developed using RStudio's <a href="http://rmarkdown.rstudio.com">R Markdown</a> and John D Blischak's <a href="https://github.com/jdblischak/workflowr">workflowr</a> package. 
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
