<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="viewport" content="width=device-width, initial-scale=1">



<title>Symbol calling and Consensus sequence</title>

<link href="site_libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>




<link rel="stylesheet" href="cosmo.css" type="text/css" />

</head>

<body>


<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <link href="https://fonts.googleapis.com/css?family=Open+Sans:100,300,400,600" rel="stylesheet" type="text/css">
    <link href="../docs/cosmo.css" rel="stylesheet">
    <link href="../docs/cosmo.min.css" rel="stylesheet">
</head>
<script src="https://ajax.googleapis.com/ajax/libs/jquery/1.12.4/jquery.min.js"></script>
<script src="../docs/bootstrap/js/bootstrap.min.js"></script>

<body>

<div class = "nav" align = "middle">
<div class= "p1"></div>
<p>
  <a class="btn btn-large btn btn-primary" type="button" href = "Logolas_home.html">HOME</a>
  &nbsp &nbsp
  <a class="btn btn-large btn btn-primary" type="button" href = "Gallery.html">GALLERY</a>
  &nbsp &nbsp
  <a class="btn btn-large btn btn-primary" type="button" href = "https://github.com/kkdey/Logolas">SOFTWARE</a>
  &nbsp &nbsp
  <a class="btn btn-large btn btn-primary" type="button" href = "workflow.html">VIGNETTES</a>
  &nbsp &nbsp
  <a class="btn btn-large btn btn-primary" type="button" href = "license.html">LICENSE</a>
   &nbsp &nbsp
  <a class="btn btn-large btn btn-primary" type="button" href = "https://github.com/kkdey/Logolas-pages">GITHUB</a>
 </p>
</div>
<!-- <div class="nav">
  <a href="index.html">Home page</a> &nbsp &nbsp
  <a href="musings.html">Musings</a> &nbsp &nbsp
  <a href="town.html">My town</a> &nbsp &nbsp
  <a href="links.html">Links</a>
</div> -->

<!-- Main content -->
</body>
</html>


<h1 class="title toc-ignore">Symbol calling and Consensus sequence</h1>


<div id="TOC">
<ul>
<li><a href="#session-information">Session information</a></li>
</ul>
</div>

<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
<!-- Update knitr chunk options -->
<!-- Insert the date the file was last updated -->
<p><strong>Last updated:</strong> 2017-09-21</p>
<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
<p><strong>Code version:</strong> 56a94be</p>
<!-- Add your analysis here -->
<p>Using the negative Logolas plot repreentation, the user can get a consensus sequence of the logos along the positions. For the DNA sequence data, the consensus sequence is usually coded by the IUPAC code, in which the bases are represented by the first letters of their names: [G]uanine, [C]ytosine, [A]denine, and [T]hymine (check the <code>getIUPAC</code> in the <a href="https://github.com/chandlerzuo/atSNP">atSNP</a> package). For the amino acid sequence, the Prosite syntax is commonly used.</p>
<p>Based on the logo heights from the Negative Logolas plot, we develop an algorithm to generate a consensus sequence, which can be used for prediction purposes to check if a given sequence matches with the underlying PWM matrix. Unlike the IUPAC and the Prosite nomenclatures that only decide on the consensus sequence using logo enrichment, the Logolas nomenclature uses both enrichment and depletion of the symbols.</p>
<p>In the Logolas nomenclature, we use uppercase of the symbol to depict enrichment and lowercase of a symbol to depict depletion of that symbol. We present an example below to demonstrate how Logolas nomenclature can be used.</p>
<p>Compared with traditional way to get the consensus sequence for nucleotide sequence, for example the <code>GetIUPACSequence</code> function from the <code>atSNP</code> package, our new method gives a better representation of the sequence.</p>
<pre class="r"><code>library(Logolas)
library(atSNP)

pwm=matrix(c(0.8,0.1,0.1,0,0.9,0.1,0,0,0.9,0.05,0.05,0,0.5,0.4,0,0.1,0.6,0.4,0,0,0.4,0.4,0.1,0.1,0.5,0,0.2,0.3,0.35,0.35,0.06,0.24,0.4,0.3,0.2,0.1,0.4,0.2,0.2,0.2,0.28,0.24,0.24,0.24,0.5,0.16,0.17,0.17,0.6,0.13,0.13,0.14,0.7,0.15,0.15,0),nrow = 4,byrow = F)
rownames(pwm)=c(&#39;A&#39;,&#39;C&#39;,&#39;G&#39;,&#39;T&#39;)
colnames(pwm)=1:ncol(pwm)

GetIUPACSequence(t(pwm))</code></pre>
<pre><code>[1] &quot;AAAMMMWMMAAAAA&quot;</code></pre>
<pre class="r"><code>GetConsensusSeq(pwm)</code></pre>
<pre><code>[1] &quot;A A A (AC) (AC) (AC) c g N N N A A (At)&quot;</code></pre>
<p>The way we give the consensus sequence is intuitive if we have a look at the logo plot.</p>
<pre class="r"><code>color_profile=list(&quot;type&quot; = &quot;per_row&quot;,&quot;col&quot; = RColorBrewer::brewer.pal(4,name =&quot;Spectral&quot;))
logomaker(pwm,color_profile = color_profile,frame_width = 1)</code></pre>
<p><img src="figure/nomenclature.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>nlogomaker(pwm,logoheight = &#39;log&#39;,color_profile = color_profile,frame_width = 1)</code></pre>
<p><img src="figure/nomenclature.Rmd/unnamed-chunk-2-2.png" width="672" style="display: block; margin: auto;" /></p>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<!-- Insert the session information into the document -->
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.3.3 (2017-03-06)
Platform: x86_64-apple-darwin13.4.0 (64-bit)
Running under: macOS Sierra 10.12.5

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
 [1] grid      stats4    parallel  stats     graphics  grDevices utils    
 [8] datasets  methods   base     

other attached packages:
 [1] atSNP_1.0            motifStack_1.18.0    Biostrings_2.42.1   
 [4] XVector_0.14.1       ade4_1.7-6           MotIV_1.30.0        
 [7] grImport_0.9-0       XML_3.98-1.9         GenomicRanges_1.26.4
[10] GenomeInfoDb_1.10.3  IRanges_2.8.2        S4Vectors_0.12.2    
[13] BiocGenerics_0.20.0  doParallel_1.0.10    iterators_1.0.8     
[16] foreach_1.4.3        data.table_1.10.4    Rcpp_0.12.12        
[19] Logolas_1.1.2       

loaded via a namespace (and not attached):
 [1] RColorBrewer_1.1-2         plyr_1.8.4                
 [3] git2r_0.18.0               bitops_1.0-6              
 [5] tools_3.3.3                zlibbioc_1.20.0           
 [7] digest_0.6.12              evaluate_0.10             
 [9] lattice_0.20-34            BSgenome_1.42.0           
[11] Matrix_1.2-10              yaml_2.1.14               
[13] LaplacesDemon_16.0.1       seqLogo_1.40.0            
[15] rtracklayer_1.34.2         stringr_1.2.0             
[17] knitr_1.16                 rprojroot_1.2             
[19] Biobase_2.34.0             BiocParallel_1.8.2        
[21] rGADEM_2.22.0              rmarkdown_1.6             
[23] magrittr_1.5               scales_0.4.1              
[25] GenomicAlignments_1.10.1   Rsamtools_1.26.2          
[27] backports_1.0.5            codetools_0.2-15          
[29] htmltools_0.3.6            SummarizedExperiment_1.4.0
[31] colorspace_1.3-2           stringi_1.1.5             
[33] munsell_0.4.3              RCurl_1.95-4.8            
[35] SQUAREM_2016.8-2          </code></pre>
</div>

<hr>
<p>
    This webpage has been developed using RStudio's <a href="http://rmarkdown.rstudio.com">R Markdown</a> and John D Blischak's <a href="https://github.com/jdblischak/workflowr">workflowr</a> package. 
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
