---
title: "String Logos"
---

<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
```{r read-chunk, include=FALSE, cache=FALSE}
knitr::read_chunk("chunks.R")
```

<!-- Update knitr chunk options -->
```{r knitr-opts-chunk, include=FALSE}
```

<!-- Insert the date the file was last updated -->
```{r last-updated, echo=FALSE, results='asis'}
```

<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
```{r code-version, echo=FALSE, results='asis'}
```

<!-- Add your analysis here -->

In this chapter, we highlight another key feature of **Logolas** - the ability to plot symbols not just for A, C, G and T, or  English alphabets, but for any alphanumeric string. 


## Character Symbol Logos

### Protein Example 

We first present an example of visualizing protein binding sites using **Logolas** where each character is represented by an English alphabet. 
Keeping with the flexibility to use any background, for the protein data example here, the background `bg` is set to uniform in panel (a) and to the amino acid composition from BLOSUM62 in panel (b).

```{r,message=FALSE,warning=FALSE, fig.height=5,fig.width=9}
library(Logolas)
library(grid)
counts_mat <- rbind(c(0, 0, 100, 1, 2), c(4, 3, 30, 35, 2),
c(100, 0, 10, 2, 7),rep(0,5),
c(4, 2, 3, 7, 70), c(1, 8, 0, 60, 3),
rep(0, 5), c(4, 2, 100, 1, 1),
c(12, 8, 16, 7, 20), c(55, 0, 1, 0, 12),
rep(0,5), c(rep(0,3), 20, 0),
rep(0,5), c(0, 0, 30, 0, 22),
c(1, 0, 12, 3, 10), rep(0,5),
c(0, 1, 0, 34, 1), c(0, 1, 12, 35, 1),
c(0, 30, 1, 10, 2), c(0, 1, 4, 100, 2))

rownames(counts_mat) <- c("A", "R", "N", "D","C", "E", "Q", "G",
"H", "I", "L", "K", "M", "F", "P", "S",
"T", "W", "Y", "V")
colnames(counts_mat) <- c("Pos 1", "Pos 2", "Pos 3", "Pos 4", "Pos 5")
cols1 <- c(rev(RColorBrewer::brewer.pal(12, "Paired"))[c(3,4,7,8,11,12,5,6,9,10)],
RColorBrewer::brewer.pal(12, "Set3")[c(1,2,5,8,9)],
RColorBrewer::brewer.pal(9, "Set1")[c(9,7)],
RColorBrewer::brewer.pal(8, "Dark2")[c(3,4,8)])
color_profile <- list("type" = "per_row",
"col" = cols1)


get_viewport_logo(1, 2)

seekViewport(paste0("plotlogo", 1))

logomaker(counts_mat,
color_profile = color_profile,
newpage = FALSE,
pop_name = "Uniform Background",
frame_width = 1)

seekViewport(paste0("plotlogo", 2))

logomaker(counts_mat,
color_profile = color_profile,
bg=c(0.074,0.052,0.045,0.054,0.025,0.034,0.054,0.074,0.026,0.068,0.099,0.058,0.025,0.047,0.039,0.057,0.051,0.013,0.034,0.073),
newpage = FALSE,
pop_name = 'BLOSUM background',
frame_width = 1)
```

### PSSM plot

However for protein motif data, PWM is not always reported. Instead a derived score matrix called PSSM (Position specific scoring matrix) is often reported.

Logolas provides a function `logo_pssm` to plot the sequence logo based on the PSSM data. One can also apply median adjustment to the PSSM matrix like in \textit{EDLogo} plot.  

```{r,message=FALSE,warning=FALSE,echo=FALSE}
bg=c(0.074,0.052,0.045,0.054,0.025,0.034,0.054,0.074,0.026,0.068,0.099,0.058,0.025,0.047,0.039,0.057,0.051,0.013,0.034,0.073)
readprotein=function(file,skip=3,nrows,nsites,bg=bg,adash,mat){
  rawdata=read.table(file = file,skip = skip,nrows = nrows)
  pssm=as.matrix(rawdata[,3:22])
  colnames(pssm)=c('A' ,'R','N','D',   'C' ,  'Q',   'E' ,  'G',   'H' ,  'I',   'L' , 'K'  , 'M' ,  'F',   'P' ,  'S'  , 'T' ,  'W',   'Y',   'V')
  rownames(pssm)=1:nrow(pssm)
  return(t(pssm))
}
pssm= readprotein('http://caps.ncbs.res.in/cgi-bin/mini/databases/3pfdb/3pfdb_pssm_download.cgi?id=PF07966&data_dir=SDB_folder',nrows = 29,nsites = 85,adash = FALSE,mat='pssm')
pssm=pssm[,19:27]
colnames(pssm)=1:9
```

```{r message = FALSE, warning = FALSE}

```

```{r,message=FALSE,warning=FALSE, fig.height=5,fig.width=9}
get_viewport_logo(1, 2)
seekViewport(paste0("plotlogo", 1))
logo_pssm(pssm,
          color_profile = color_profile,
          control = list(quant = 0, gap_ylab = 3.5, round_off = 0),
          newpage = FALSE,
          pop_name = "PSSM : no adj.")

seekViewport(paste0("plotlogo", 2))
logo_pssm(pssm,
          color_profile = color_profile,
          control = list(quant = 0.5, gap_ylab = 3.5, round_off = 0),
          newpage = FALSE,
          pop_name = "PSSM: median adj.")


```

## Strings +  Character Symbols Logos

### Mutation Signatures plot

Next we present an example of cancer mutation signature profiles where string symbols are mixed with character symbols. 

A cancer mutation signature is usually represented by the type of mutation (C->T, C->A, C->G, T->A, T->C.T->G) along with the immediately surrounding left and right flanking bases, each of which can take the symbols A, C, G and T. We use the string symbol feature of **Logolas** to represent the substitutions $X\rightarrow Y$ and the character symbols for the flanking bases.

User may want to have the C in $C>T$ to be of the same color as the C at the flanking bases and Logolas coloring profile provides the user the flexibility to color each symbol instead of a string. While `per_row` profile would assign a separate color to $C->T$ and the standalone base $C$ at the flanking bases, the `per_symbol` option can actually recognise each character in the string and assign a color to it. We see the difference through the example below. 


```{r,message=FALSE, fig.height=7,fig.width=9}
mFile <- system.file("Exfiles/pwm1", package="seqLogo")
m <- read.table(mFile)
p <- seqLogo::makePWM(m)
pwm_mat <- slot(p,name = "pwm")
mat1 <- cbind(pwm_mat[,c(3,4)], rep(NA,4), pwm_mat[,c(5,6)]);
colnames(mat1) <- c("-2", "-1", "0", "1", "2")
mat2 <- cbind(rep(NA,6), rep(NA,6),
c(0.5, 0.2, 0.2, 0.05, 0.05, 0),
rep(NA,6), rep(NA,6))
rownames(mat2) <- c("C>T", "C>A", "C>G",
"T>A", "T>C", "T>G")
table <- rbind(mat1, mat2)

get_viewport_logo(1, 2)

seekViewport(paste0("plotlogo", 1))
color_profile <- list("type" = "per_row",
"col" = RColorBrewer::brewer.pal(dim(table)[1],name ="Spectral"))
nlogomaker(table,
          color_profile = color_profile,
          logoheight = "log",
          pop_name = "color per element",
          newpage = FALSE)


seekViewport(paste0("plotlogo", 2))
cols = RColorBrewer::brewer.pal.info[RColorBrewer::brewer.pal.info$category == 'qual',]
col_vector = unlist(mapply(RColorBrewer::brewer.pal, cols$maxcolors, rownames(cols)))

total_chars = c("A", "B", "C", "D", "E", "F", "G", "H", "I", "J", "K", "L", "M", "N", "O",
"P", "Q", "R", "S", "T", "U", "V", "W", "X", "Y", "Z", "zero", "one", "two",
"three", "four", "five", "six", "seven", "eight", "nine", "dot", "comma",
"dash", "colon", "semicolon", "leftarrow", "rightarrow")

set.seed(20)
color_profile <- list("type" = "per_symbol",
"col" = sample(col_vector, length(total_chars), replace=FALSE))

nlogomaker(table,
          color_profile = color_profile,
          logoheight = "log",
          total_chars = total_chars,
          pop_name = "color per character",
          newpage = FALSE)
```

Another coloring option available in **Logolas** is `per_column`, in which we assign a specific color for a specific column. This sort of coloring might be useful when the user wants to highlight the difference between columns [Check Gallery example of Birds data]


## Full string plots

Next we consider an example where all the symbols are alphanumeric strings. 

### Histone mark composition

We apply Logolas on an example data from [Koch et al (2007)](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1891331/) that summarizes the relative abundance patterns 5 different histone marks in different genomic contexts - intergenic sequence, intron, exon, gene start and gene end for the lymphoblastoid cell line, GM06990, in the ChIP-CHIP data. Note here that the histone marks are alphanumeric symbols, for example, H3K4ME1. we use as background a table of relative abundances generated from randomly simulated data in Koch *et al* (2007).

```{r,message=FALSE, fig.height=5,fig.width=9}
mat <- rbind(c(326, 296, 81, 245, 71),
             c(258, 228, 55, 273, 90),
             c(145, 121, 29, 253, 85),
             c(60, 52, 23, 180, 53),
             c(150, 191, 63, 178, 63))

bgmat <- rbind(c(542, 218, 118, 108, 33),
               c(480, 204, 107, 87, 26),
               c(346, 131, 71, 66, 20),
               c(199, 72, 41, 43, 13),
               c(368, 101, 67, 80, 30))

rownames(mat) <- c("H3K4ME1", "H3K4ME2", "H3K4ME3", "H3AC", "H4AC")
colnames(mat) <- c("Intergenic","Intron","Exon",
                   "Gene start ","Gene end")

set.seed(201)
color_profile <- list("type" = "per_row",
                      "col" = sample(RColorBrewer::brewer.pal(10,name = "Spectral"), dim(mat)[1]))

get_viewport_logo(1, 2)

seekViewport(paste0("plotlogo", 1))
Logolas::logomaker(mat,
          color_profile = color_profile,
          pop_name = "",
          newpage = FALSE,
          col_line_split = "black",
          control = list(gap_ylab = 3.5,
                         totbins = 3))

seekViewport(paste0("plotlogo", 2))
Logolas::nlogomaker(mat,
                    logoheight = 'log',
                    bg = bgmat,
                    pop_name = "",
                    color_profile = color_profile,
                    newpage = FALSE,
                    control = list(gap_ylab=3.5,
                                   round_off = 1,
                                   posbins = 3, negbins = 3))

```

The control options `posbins`,  `negbins` and `totbins` determine the number of ticks in the positive and negative parts of the Y axis in the logo plot, whereas `gap_ylab` denotes the spacing between the Y axis label and the y axis ticks. 



## Session information

<!-- Insert the session information into the document -->
```{r session-info}
```
