---
title: " Why Logolas? "
---

<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
```{r read-chunk, include=FALSE, cache=FALSE}
knitr::read_chunk("chunks.R")
```

<!-- Update knitr chunk options -->
```{r knitr-opts-chunk, include=FALSE}
```

<!-- Insert the date the file was last updated -->
```{r last-updated, echo=FALSE, results='asis'}
```

<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
```{r code-version, echo=FALSE, results='asis'}
```

<!-- Add your analysis here -->

Ever since its introduction due to Schneider & Stephens (1990), sequence logos have proved to be an effective tool in informative visualization of patterns in aligned sets of sequences. There are several existing packages for plotting sequence logos, like 

- [seqLogo](http://bioconductor.org/packages/release/bioc/html/seqLogo.html) (exclusively for DNA sequence motif visualization)
- [seq2Logo](http://www.cbs.dtu.dk/biotools/Seq2Logo/) (amino acid binding motifs visualization)
- [WebLogo](http://weblogo.berkeley.edu/logo.cgi) (web based tool for logo plotting)
- [motifStack](http://bioconductor.org/packages/release/bioc/html/motifStack.html)


We introduce another Logo visualization package called [Logolas](https://bioconductor.org/packages/release/bioc/html/Logolas.html) which addresses some limitations of the above packages and makes logo visualization a more generic tool with potential applications in a much wider scope of problems, in comparison to the existing softwares. We highlight some of these advantages below.


- **Strings as symbols** : General logo building softwares have limited library of symbols (A, C, G and T for seqLogo, alphabets for WebLogo and seq2Logo). However Logolas lets the user plot, besides the usual alphabetic symbols, any alphanumeric string, comprising of English alphabets, numbers, punctuation marks, arrows etc. It also provides an easy interface for the user to create her own logo and add to the library of symbols that can be plotted (check the Vignette).


- **More parsimonious Representation** : General logo plotting softwares highlight only enrichment of certain symbols, but Logolas allows the user to highlight both enrichment and depletion of symbols at any position, leading to more parsimonious and visually appealing representation. 


- **Symbol calling Nomenclature** : Based on enrichment and depletion of symbols, Logolas introduces a new nomenclature to call symbols at each position. The Logolas nomenclature is more informative and generic compared to the IUPAC nomenclature for DNA sequence binding motifs and Prosite nomenclature for protein binding motifs.


- **Dirichlet Adaptive Shrinkage** : Most existing logo plotting tools take position weight matrix (PWM) as input for the logo plot. But they tend to effectively adapt to the scale of the underlying positional frequencies from which the PWM is created. In Logolas, we provide a Dirichlet Adaptive Shrinkage method (dash) approach for adaptively scaling the heights of the logos based on the frequency scale. This approach is similar in lines to the **ashr** software due to M. Stephens (2016).


- **Better stylizations** : Logolas also allows for new and flexible stylizations of the logos, involving color palettes, fill and border styles, in the choice of entropy for determining heights of the logos etc. In our vignettes, we also show how to plot multi-panel Logolas plots and combine Logolas plots with ggplot2 and base R graphics.



#Reference

Schneider, T. D., & Stephens, R. M. (1990). Sequence logos: a new way to display consensus sequences. Nucleic acids research, 18(20), 6097-6100.

Stephens, M. (2016). False discovery rates: a new deal. Biostatistics, 18(2), 275-294.

## Session information

<!-- Insert the session information into the document -->
```{r session-info}
```
